src/main/kotlin/com/demonwav/mcdev/platform/mixin/inspection/MixinClassReferenceInspection.kt

```diff
@@ -41,7 +41,7 @@ class MixinClassReferenceInspection : MixinInspection() {
                 return
             }
 
-            holder.registerProblem(type, "${referencedClass.name} is a Mixin class, and cannot be referenced directly")
+            holder.registerProblem(type, "Mixin class cannot be referenced directly")
```

Comment:
For your reference then:

*Accessor Mixins* are the only case currently where it's legal to refer to a Mixin class *outside* of the mixin hierarchy and they can be identified by:

* Mixin itself must be an `interface`
* All mixin targets must be `class`es
* All member methods must be decorated with `@Accessor` or `@Invoker`

If the above conditions are met then the Mixin is applied as an *Accessor Mixin* and is allowed to be referenced from consumer code:

```java
@Mixin(Entity.class)
public interface MixinEntity {

    @Accessor
    public abstract Set<String> getTags();

}
```

In consumer code:

```java
Entity entity = this.getSomeEntity();
// Note that this cast is legal because the mixin is an accessor
Set<String> entityTags = ((MixinEntity)entity).getTags();
```